cpg_workflows/workflow.py

```diff
@@ -41,6 +41,35 @@
 TargetT = TypeVar('TargetT', bound=Target)
 
 
+def path_walk(expected, collected: set) -> set:
+    """
+    recursive walk of expected_out
+    if the object is iterable, walk it
+    this gets around the issue with nested lists and dicts
+    mainly around the use of Array outputs from Cromwell
+
+    Args:
+        expected (Any): any type of object containing Paths
+        collected (set): all collected paths so far
+
+    Returns:
+        a list of all collected nodes
```

Comment:
Added a test and a docstring example.

I've also added a breaking statement so if two different files would be written to the same output path it will raise an Exception. That's based on an issue where I accidentally created a stage definition that wrote 2 files to the same location, causing issues downstream as they were overwritten with only a single file's content